src/main/java/teammates/storage/api/FeedbackSessionsDb.java

```diff
@@ -544,6 +551,26 @@ public void deleteFeedbackSessionsForCourses(List<String> courseIds) {
     }
     
     @SuppressWarnings("unchecked")
+    private List<FeedbackSession> getFeedbackSessionEntitiesNeedingClosingEmail() {
+        Query q = getPm().newQuery(FeedbackSession.class);
+        q.declareParameters("boolean sentParam, boolean enableParam, Enum notTypeParam");
+        q.setFilter("sentClosingEmail == sentParam && isClosingEmailEnabled == enableParam "
+                    + "&& feedbackSessionType != notTypeParam");
+        
+        return (List<FeedbackSession>) q.execute(false, true, FeedbackSessionType.PRIVATE);
+    }
+    
+    @SuppressWarnings("unchecked")
+    private List<FeedbackSession> getFeedbackSessionEntitiesNeedingClosedEmail() {
+        Query q = getPm().newQuery(FeedbackSession.class);
+        q.declareParameters("boolean sentParam, boolean enableParam, Enum notTypeParam");
+        q.setFilter("sentClosedEmail == sentParam && isClosedEmailEnabled == enableParam "
+                    + "&& feedbackSessionType != notTypeParam");
```

Comment:
Adding a new a attribute is a lot more complicated as it needs to be kept consistent when the session deadlines are changed. Furthermore, there is a risk that we'll end up sending the reminder repeatedly if something goes wrong with setting the flag. I don't mind doing it if it can be done in a fool proof way, but I think the quickest solution is to narrow down the result set into last 48 hours.